storm-core/src/jvm/backtype/storm/scheduler/resource/strategies/scheduling/DefaultResourceAwareStrategy.java

```diff
@@ -97,7 +101,7 @@ public ResourceAwareStrategy(Cluster cluster, Topologies topologies) {
 
         if (spouts.size() == 0) {
             LOG.error("Cannot find a Spout!");
-            return null;
+            return SchedulingResult.failure(SchedulingStatus.FAIL_INVALID_TOPOLOGY, "Cannot find a Spout!");
```

Comment:
Yeah, the DefaultTopologyValidator we have [does nothing](https://github.com/apache/storm/blob/418792f84c95962a5f2e8d8f98aab73e8138c31a/storm-core/src/jvm/backtype/storm/nimbus/DefaultTopologyValidator.java#L28-L30).  I think it might be good to have a new Jira that adds a check to the DefaultValidator that at least one spout exists, and also removes code from this scheduler that repeats the same check.